networking_oneview/ml2/drivers/oneview/common.py

```diff
@@ -540,3 +585,114 @@ def remove_inconsistence_from_db(
     database_manager.delete_oneview_network_lig(
         session, oneview_network_id=oneview_network_id
     )
+
+
+def check_valid_resources():
+    """Verify if the OneView resources exist.
+
+    Verify if the resources described on the configuration file
+    exist on OneView.
+
+    :raise OneViewResourceNotFoundException: If any of the OneView
+        resources does not exist.
+    :raise ElementNotFoundException: If the UplinkSet name is not
+        in the LIG's UplinkSets list. If a Network is not associated
+        to any UplinkSet.
+    """
+    check_uplinkset_mappings_resources()
+    check_flat_net_mappings_resources()
+
+
+def check_uplinkset_mappings_resources():
+    """Verify if the Logical Interconnect Groups and UplinkSets exist.
+
+    :raise OneViewResourceNotFoundException:: If a Logical Interconnect
+        Group does not exist.
+    :raise ElementNotFoundException: If the UplinkSet name is not in
+        the LIG's UplinkSets list.
+    """
+    mappings = load_conf_option_to_dict(CONF.oneview.uplinkset_mappings)
+
+    for physnet in mappings:
+        # NOTE(mrtenio): Verifying if the providers exist is necessary?
+        provider = zip(
+            mappings.get(physnet)[0::2],
+            mappings.get(physnet)[1::2])
+
+        # Check if Logical Interconnect Groups and UplinkSets  exist
+        for lig_id, uplinkset_name in provider:
+            try:
+                lig = get_logical_interconnect_group_by_id(lig_id)
+            except exceptions.OneViewResourceNotFoundException as err:
+                raise err
+
+            uplinksets = lig.get('uplinkSets')
+
+            try:
+                get_uplinkset_by_name_from_list(uplinksets, uplinkset_name)
+            except exceptions.ElementNotFoundException as err:
+                raise err
+
+
+def check_flat_net_mappings_resources():
+    """Verify if the Ethernet Networks exist.
+
+    :raise OneViewResourceNotFoundException: If an
+        Ethernet Network does not exist.
+    :raise ElementNotFoundException: If there is no UplinkSet
+        associated with the Network.
+    """
+    mappings = load_conf_option_to_dict(CONF.oneview.flat_net_mappings)
+
+    for physnet in mappings:
+        oneview_network_ids = mappings.get(physnet)
```

Comment:
Can we have more than one with the same physnet??? I'm not sure about this being a list of network ids...

Reply:
I am iterating because that is how the mappings is structured. If we only map Flat networks in the flat_net_mappings, there should not be two networks for the same physical network.